src/k8s/pkg/utils/cidr.go

```diff
@@ -4,9 +4,43 @@ import (
 	"fmt"
 	"math/big"
 	"net"
+	"strconv"
 	"strings"
+
+	"github.com/canonical/lxd/lxd/util"
 )
 
+// findMatchingNodeAddress returns the IP address of a network interface that belongs to the given CIDR.
+func findMatchingNodeAddress(cidr *net.IPNet) (net.IP, error) {
+	addrs, err := net.InterfaceAddrs()
+	if err != nil {
+		return nil, fmt.Errorf("could not get interface addresses: %w", err)
+	}
+
+	var selectedIP net.IP
+	selectedSubnetBits := 32
+
+	for _, addr := range addrs {
+		ipNet, ok := addr.(*net.IPNet)
+		if !ok {
+			continue
+		}
+		if cidr.Contains(ipNet.IP) {
+			if _, subnetBits := cidr.Mask.Size(); subnetBits < selectedSubnetBits {
```

Comment:
note that this works for IPv4 address, but will probably break in IPv6 (since addresses there can go up to `/128`).

An alternative would be to set this initially to `-1`, then update the check to `selectedSubnetBits == -1 || subnetBits < selectedSubnetBits`

Reply:
Thanks for the hint, I've updated the PR with a new check for the Subnet bits.